bin/hh-autoload.hack

```diff
@@ -116,12 +122,21 @@ final class GenerateScript {
   private static function checkRoot(): void {
     // NO HSL HERE - autoloader is not yet initialized
     if (!\file_exists('hh_autoload.json')) {
-      \fwrite(
-        \STDERR,
-        "This executable must be ran from a directory containing an ".
-        "hh_autoload.json\n",
-      );
-      exit(1);
+      if (!static::nativeAutoloadingIsBeingUsed()) {
+        \fwrite(
+          \STDERR,
+          "This executable must be ran from a directory containing an ".
+          "hh_autoload.json\n",
+        );
+        exit(1);
+      } else {
```

Comment:
Small style nit: remove the else

When an if or similar ends with exit/return/throw, we prefer to make the following code unnested - for more complicated branching, it greatly helps with readability

Reply:
Agreed
```
            }
          }
        }
      }
    }
  }
}
```